src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ErrorNotificationHandlerTest.java

```diff
@@ -0,0 +1,249 @@
+
+        // and
+        String output = outputCapture.toString();
+
+        assertThat(output).containsPattern("INFO.+"
+            + ErrorNotificationHandler.class.getCanonicalName()
+            + ":\\d+: Processing error notification for "
+            + ZIP_FILE_NAME
+        );
+        assertThat(output).containsPattern("ERROR \\[error-notification-handler\\] "
+            + ErrorNotificationExceptionHandler.class.getCanonicalName()
+            + ":\\d+: Client error. Dead lettering message \\(ID: "
+            + MESSAGE_ID
+            + "\\)\n"
+        );
+        assertThat(output).containsPattern("Caused by: "
+            + ErrorNotificationException.class.getCanonicalName()
+            + ": "
+            + HttpClientErrorException.class.getCanonicalName()
+            + ": 400 Bad Request\n"
+        );
+    }
+
+    @Test
+    public void should_complete_the_message_when_success_received_from_notification_client()
+        throws JsonProcessingException {
+        // given
+        IMessage message = getServiceBusMessage();
+
+        // and
+        given(notificationClient.notify(any(ErrorNotificationRequest.class)))
+            .willReturn(new ErrorNotificationResponse(NOTIFICATION_ID));
+        given(queueClient.completeAsync(any(UUID.class))).willReturn(EMPTY_FUTURE);
+
+        // when
+        CompletableFuture<Void> future = notificationHandler.onMessageAsync(message);
+        future.join();
+
+        // then
+        assertThat(future.isCompletedExceptionally()).isFalse();
+
+        // and
+        String output = outputCapture.toString();
+
+        assertThat(output).containsPattern("INFO.+"
+            + ErrorNotificationHandler.class.getCanonicalName()
+            + ":\\d+: Processing error notification for "
+            + ZIP_FILE_NAME
+        );
+        assertThat(output).containsPattern("INFO  \\[error-notification-service\\] "
+            + ErrorNotificationService.class.getCanonicalName()
+            + ":\\d+: Error notification for "
+            + ZIP_FILE_NAME
+            + " published. ID: "
+            + NOTIFICATION_ID
+        );
+    }
+
+    private IMessage getServiceBusMessage() throws JsonProcessingException {
+        ProcessEvent event = eventRepository.save(
+            new ProcessEvent("container_name", ZIP_FILE_NAME, Event.DOC_FAILURE)
+        );
+
+        IMessage message = spy(new Message(
+            MESSAGE_ID,
+            OBJECT_MAPPER.writeValueAsBytes(
+                new ErrorMsg(
+                    "id",
+                    event.getId(),
+                    ZIP_FILE_NAME,
+                    "jurisdiction",
+                    "po_box",
+                    null,
+                    ERR_FILE_LIMIT_EXCEEDED,
+                    "description"
+                )
+            ),
+            MediaType.APPLICATION_JSON_UTF8_VALUE
+        ));
+
+        given(message.getLockToken()).willReturn(UUID.randomUUID());
```

Comment:
worth putting these two given will return outside of this method

Reply:
not sure how it's worth. these are dedicated to activity caused within servicebus for which we have no control of. there is no reason to verify them just make sure source code doesn't crash with null pointers